### ISSUE TITLE
Error building APK: "Missing 'name' key attribute on element activity at AndroidManifest.xml"

### ISSUE THREAD

**Issue by npetitclerc** [kivy/python-for-android#1979]
### Versions

* Python: 3.7
* OS: masOS Mojave 10.14.6
* p4a: Master branch version, installed via 
pip install git+https://github.com/kivy/python-for-android.git
2019.8.9.1.dev0-cp37-none-any.whl
* Cython: 0.29.13

### Description

I want to build a apk file for the Pong tutorial, but it crashes with the error:
"Missing 'name' key attribute on element activity at AndroidManifest.xml:54:9-68:24"
and indeed the AndroidManifest.html file is missing that parameter.
 
### To reproduce:
> virtualenv --python=/usr/local/opt/python/bin/python3.7 ~/kivy_venv5
> source ~/kivy_venv5/bin/activate
> pip install git+https://github.com/kivy/python-for-android.git
> pip3 install --upgrade cython
> pip3 install --upgrade kivy

Following instal instructions from: https://python-for-android.readthedocs.io/en/latest/quickstart/#installation
> touch ~/.android/repositories.cfg
> /Users/nic/Library/Android/sdk/tools/bin/sdkmanager "platforms;android-27”
> /Users/nic/Library/Android/sdk/tools/bin/sdkmanager "build-tools;26.0.2"

Add to ~/.bashrc
export ANDROIDSDK="/Users/nic/Library/Android/sdk"
export ANDROIDNDK=/Users/nic/projects/kivy/NDK/android-ndk-r20
export ANDROIDAPI="29"  # Target API version of your application
export NDKAPI="21"  # Minimum supported API version of your application

Source ~/.bashrc

p4a apk --private /Users/nic/projects/kivy/pong5/ --package=org.example.mypong5 --name "My Pong5" --version 0.1 --bootstrap=sdl2 --requirements=python3,kivy --debug 2>&1 | tee out.log

### Logs
[INFO]:    Will compile for the following archs: armeabi-v7a
[INFO]:    Found Android API target in $ANDROIDAPI: 29
[INFO]:    Available Android APIs are (27, 29)
[INFO]:    Requested API target 29 is available, continuing.
[INFO]:    Found NDK dir in $ANDROIDNDK: /Users/nic/projects/kivy/NDK/android-ndk-r20
[INFO]:    Found NDK version 20
[INFO]:    Found Android API target in $NDKAPI
[INFO]:    Found virtualenv at /Users/nic/kivy_venv5/bin/virtualenv
[INFO]:    ccache is missing, the build will not be optimized in the future.
[INFO]:    Found the following toolchain versions: ['4.9']
[INFO]:    Picking the latest gcc toolchain, here 4.9
[WARNING]: Missing executable: libtoolize is not installed
[INFO]:    Of the existing distributions, the following meet the given requirements:
[INFO]:    	unnamed_dist_17: min API 21, includes recipes (hostpython3, libffi, openssl, sdl2_image, sdl2_mixer, sdl2_ttf, sqlite3, python3, sdl2, setuptools, six, pyjnius, android, kivy), built for archs (armeabi-v7a)
[INFO]:    unnamed_dist_17 has compatible recipes, using this one
[INFO]:    Of the existing distributions, the following meet the given requirements:
[INFO]:    	unnamed_dist_17: min API 21, includes recipes (hostpython3, libffi, openssl, sdl2_image, sdl2_mixer, sdl2_ttf, sqlite3, python3, sdl2, setuptools, six, pyjnius, android, kivy), built for archs (armeabi-v7a)
[INFO]:    unnamed_dist_17 has compatible recipes, using this one
[INFO]:    -> directory context /Users/nic/.python-for-android/dists/unnamed_dist_17
Listing '/private/var/folders/z4/l0d41jtj7t3g0mqtk8kqd88w0000gn/T/p4a-extra-env-qwa3e_5i'...
Listing '/Users/nic/projects/kivy/pong5'...
Compiling '/Users/nic/projects/kivy/pong5/main.py'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/modules'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/Kivy-1.11.1-py3.7.egg-info'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/android'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/jnius'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/jnius/src'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/jnius/src/org'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/jnius/src/org/jnius'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/core'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/core/audio'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/core/camera'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/core/clipboard'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/core/gl'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/core/image'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/core/spelling'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/core/text'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/core/video'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/core/window'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/data'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/data/fonts'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/data/glsl'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/data/images'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/data/keyboards'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/data/logo'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/deps'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/effects'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/extras'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/garden'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/graphics'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/graphics/cgl_backend'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/include'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/input'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/input/postproc'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/input/providers'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/lang'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/lib'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/lib/gstplayer'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/lib/vidcore_lite'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/modules'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/network'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/storage'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/tools'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/tools/gles_compat'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/tools/highlight'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/tools/packaging'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/tools/packaging/pyinstaller_hooks'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/tools/pep8checker'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/tools/theming'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/tools/theming/defaulttheme'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/uix'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/uix/behaviors'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/kivy/uix/recycleview'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/pkg_resources'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/pkg_resources/_vendor'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/pkg_resources/_vendor/packaging'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/pkg_resources/extern'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/pyjnius-1.2.1.dev0-py3.7.egg-info'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/setuptools'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/setuptools/_vendor'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/setuptools/_vendor/packaging'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/setuptools/command'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/setuptools/extern'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/setuptools-40.9.0-py3.7.egg-info'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/six-1.10.0-py3.7.egg-info'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/usr'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/usr/local'...
Listing '/Users/nic/.python-for-android/dists/unnamed_dist_17/_python_bundle/_python_bundle/site-packages/usr/local/bin'...
[INFO]:    Selecting java build tool:
[INFO]:    Detected highest available build tools version to be 29.0.2
[INFO]:        Building with gradle, as gradle executable is present
[DEBUG]:   -> running gradlew assembleDebug
[DEBUG]:
[DEBUG]:   	> Task :processDebugManifest FAILED
[DEBUG]:   	/Users/nic/.python-for-android/dists/unnamed_dist_17/src/main/AndroidManifest.xml:54:9-68:24 Error:
[DEBUG]:   	        Missing 'name' key attribute on element activity at AndroidManifest.xml:54:9-68:24
[DEBUG]:   	/Users/nic/.python-for-android/dists/unnamed_dist_17/src/main/AndroidManifest.xml Error:
[DEBUG]:   	        Validation failed, exiting
[DEBUG]:
[DEBUG]:   	See http://g.co/androidstudio/manifest-merger for more information about the manifest merger.
[DEBUG]:
[DEBUG]:
[DEBUG]:
[DEBUG]:   	FAILURE: Build failed with an exception.
[DEBUG]:
[DEBUG]:   	* What went wrong:
[DEBUG]:   	Execution failed for task ':processDebugManifest'.
[DEBUG]:   	> Manifest merger failed with multiple errors, see logs
[DEBUG]:
[DEBUG]:   	* Try:
[DEBUG]:   	Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[DEBUG]:
[DEBUG]:   	* Get more help at https://help.gradle.org
[DEBUG]:
[DEBUG]:   	BUILD FAILED in 0s
[DEBUG]:   	11 actionable tasks: 2 executed, 9 up-to-date

Exception in thread background thread for pid 1712:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.4/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python/3.7.4/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/nic/kivy_venv5/lib/python3.7/site-packages/sh.py", line 1540, in wrap
    fn(*args, **kwargs)
  File "/Users/nic/kivy_venv5/lib/python3.7/site-packages/sh.py", line 2459, in background_thread
    handle_exit_code(exit_code)
  File "/Users/nic/kivy_venv5/lib/python3.7/site-packages/sh.py", line 2157, in fn
    return self.command.handle_command_exit_code(exit_code)
  File "/Users/nic/kivy_venv5/lib/python3.7/site-packages/sh.py", line 815, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_1:

  RAN: /Users/nic/.python-for-android/dists/unnamed_dist_17/gradlew assembleDebug

  STDOUT:

> Task :processDebugManifest FAILED
/Users/nic/.python-for-android/dists/unnamed_dist_17/src/main/AndroidManifest.xml:54:9-68:24 Error:
        Missing 'name' key attribute on element activity at AndroidManifest.xml:54:9-68:24
/Users/nic/.python-for-android/dists/unnamed_dist_17/src/main/AndroidManifest.xml Error:
        Validation failed, exiting

See http://g.co/androidstudio/manifest-merger for more information about the manifest merger.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':processDebugManifest'.
> Manifest merger failed with multiple errors, see logs

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
11 actionable tasks: 2 executed, 9 up-to-date

  STDERR:

[INFO]:    STDOUT (last 20 lines of 27):

See http://g.co/androidstudio/manifest-merger for more information about the manifest merger.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':processDebugManifest'.
> Manifest merger failed with multiple errors, see logs

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
11 actionable tasks: 2 executed, 9 up-to-date

[INFO]:    STDERR:

[INFO]:    COMMAND:
cd /Users/nic/.python-for-android/dists/unnamed_dist_17 && /Users/nic/.python-for-android/dists/unnamed_dist_17/gradlew assembleDebug

[WARNING]: ERROR: /Users/nic/.python-for-android/dists/unnamed_dist_17/gradlew failed!
WARNING: Received a --sdk argument, but this argument is deprecated and does nothing.
No compiled python is present to zip, skipping.
No setup.py/pyproject.toml used, copying full private data into .apk.
Applying Java source code patches...
Applying patch: src/patches/SDLActivity.java.patch
Warning: failed to apply patch (exit code 1), assuming it is already applied: src/patches/SDLActivity.java.patch

**Comment by npetitclerc** [comment]
FYI I managed to get around this error by using miniconda as a virtual environment. Using the the steps described here: https://github.com/kivy/python-for-android/issues/1987

However, this is NOT a proper workaround since the app compiled this way crashes on startup...

**Comment by AndreMiras** [comment]
I got this error once too, I didn't investigate further, simply removed the local `.buildozer/` directory and ran again and problem was gone.
Log was:
```
Listing '/home/andre/workspace/EtherollApp/.buildozer/android/platform/build/dists/etheroll/_python_bundle/_python_bundle/site-packages/web3-4.8.1.dist-info'...
[INFO]:    Selecting java build tool:                  
[INFO]:    Detected highest available build tools version to be 29.0.2         
[INFO]:        Building with gradle, as gradle executable is present
[DEBUG]:   -> running gradlew assembleDebug                                                         
[DEBUG]:                                          
[DEBUG]:        > Task :processDebugManifest FAILED
[DEBUG]:        /home/andre/workspace/EtherollApp/.buildozer/android/platform/build/dists/etheroll/src/main/AndroidManifest.xml:35:5-81 Warning:
[DEBUG]:                Element uses-permission#android.permission.WRITE_EXTERNAL_STORAGE at AndroidManifest.xml:35:5-81 duplicated with element declared at AndroidManifest.xml:28:5-81
[DEBUG]:        /home/andre/workspace/EtherollApp/.buildozer/android/platform/build/dists/etheroll/src/main/AndroidManifest.xml:62:9-76:24 Error:
[DEBUG]:                Missing 'name' key attribute on element activity at AndroidManifest.xml:62:9-76:24
[DEBUG]:        /home/andre/workspace/EtherollApp/.buildozer/android/platform/build/dists/etheroll/src/main/AndroidManifest.xml Error:
[DEBUG]:                Validation failed, exiting
[DEBUG]:
[DEBUG]:        See http://g.co/androidstudio/manifest-merger for more information about the manifest merger.
[DEBUG]:
[DEBUG]:
[DEBUG]:
[DEBUG]:        FAILURE: Build failed with an exception.
[DEBUG]:
[DEBUG]:        * What went wrong:
[DEBUG]:        Execution failed for task ':processDebugManifest'.
[DEBUG]:        > Manifest merger failed with multiple errors, see logs
[DEBUG]:
[DEBUG]:        * Try:
[DEBUG]:        Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[DEBUG]:
[DEBUG]:        * Get more help at https://help.gradle.org
[DEBUG]:
[DEBUG]:        BUILD FAILED in 0s
[DEBUG]:        11 actionable tasks: 11 executed
[DEBUG]:
[DEBUG]:
Exception in thread background thread for pid 27213:
Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/andre/.local/lib/python3.6/site-packages/sh.py", line 1540, in wrap
    fn(*args, **kwargs)
  File "/home/andre/.local/lib/python3.6/site-packages/sh.py", line 2459, in background_thread
    handle_exit_code(exit_code)
  File "/home/andre/.local/lib/python3.6/site-packages/sh.py", line 2157, in fn
    return self.command.handle_command_exit_code(exit_code)
  File "/home/andre/.local/lib/python3.6/site-packages/sh.py", line 815, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_1:

  RAN: /home/andre/workspace/EtherollApp/.buildozer/android/platform/build/dists/etheroll/gradlew assembleDebug

  STDOUT:

> Task :processDebugManifest FAILED
/home/andre/workspace/EtherollApp/.buildozer/android/platform/build/dists/etheroll/src/main/AndroidManifest.xml:35:5-81 Warning:
        Element uses-permission#android.permission.WRITE_EXTERNAL_STORAGE at AndroidManifest.xml:35:5-81 duplicated with element declared at AndroidManifest.xml:28:5-81
/home/andre/workspace/EtherollApp/.buildozer/android/platform/build/dists/etheroll/src/main/AndroidManifest.xml:62:9-76:24 Error:
        Missing 'name' key attribute on element activity at AndroidManifest.xml:62:9-76:24
/home/andre/workspace/EtherollApp/.buildozer/android/platform/build/dists/etheroll/src/main/AndroidManifest.xml Error:
        Validation failed, exiting

See http://g.co/androidstudio/manifest-merger for more information about the manifest merger.



FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':processDebugManifest'.
> Manifest merger failed with multiple errors, see logs

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
11 actionable tasks: 11 executed




  STDERR:

[INFO]:    STDOUT (last 20 lines of 31):
[DEBUG]:See http://g.co/androidstudio/manifest-merger for more information about the manifest merger.



FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':processDebugManifest'.
> Manifest merger failed with multiple errors, see logs

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
11 actionable tasks: 11 executed

[INFO]:    STDERR:

[INFO]:    COMMAND:
cd /home/andre/workspace/EtherollApp/.buildozer/android/platform/build/dists/etheroll && /home/andre/workspace/EtherollApp/.buildozer/android/platform/build/dists/etheroll/gradlew assembleDebug

[WARNING]: ERROR: /home/andre/workspace/EtherollApp/.buildozer/android/platform/build/dists/etheroll/gradlew failed!
WARNING: Received a --sdk argument, but this argument is deprecated and does nothing.
No compiled python is present to zip, skipping.
No setup.py/pyproject.toml used, copying full private data into .apk.
Applying Java source code patches...
Applying patch: src/patches/SDLActivity.java.patch
# Command failed: /usr/bin/python3 -m pythonforandroid.toolchain apk --debug --bootstrap=sdl2 --dist_name etheroll --name Etheroll --version 2019.0911 --package com.github.andremiras.etheroll --android_api 27 --minsdk 21 --ndk-api 21 --private /home/andre/workspace/EtherollApp/.buildozer/android/app --permission INTERNET --permission WRITE_EXTERNAL_STORAGE --presplash /home/andre/workspace/EtherollApp/docs/images/etheroll-logo.png --icon /home/andre/workspace/EtherollApp/docs/images/icon.png --orientation portrait --window --service service:etherollapp/service/main.py --copy-libs --local-recipes /home/andre/workspace/EtherollApp/src/python-for-android/recipes --blacklist /home/andre/workspace/EtherollApp/blacklist.txt --arch armeabi-v7a --color=always --storage-dir="/home/andre/workspace/EtherollApp/.buildozer/android/platform/build" --ndk-api=21
# ENVIRONMENT:
#     CLUTTER_IM_MODULE = 'xim'
#     LC_ALL = 'en_US.UTF-8'
#     NVM_DIR = '/home/andre/bin/nvm-0.33.11'
#     LS_COLORS = 'rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=0
1;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:'
#     LESSCLOSE = '/usr/bin/lesspipe %s %s'
#     XDG_MENU_PREFIX = 'gnome-'
#     LANG = 'en_US.UTF-8'
#     DISPLAY = ':0'
#     GNOME_SHELL_SESSION_MODE = 'ubuntu'
#     COLORTERM = 'truecolor'
#     NVM_CD_FLAGS = ''
#     USERNAME = 'andre'
#     XDG_VTNR = '2'
#     SSH_AUTH_SOCK = '/run/user/1000/keyring/ssh'
#     MANDATORY_PATH = '/usr/share/gconf/ubuntu.mandatory.path'
#     XDG_SESSION_ID = '2'
#     USER = 'andre'
#     DESKTOP_SESSION = 'ubuntu'
#     QT4_IM_MODULE = 'xim'
#     TEXTDOMAINDIR = '/usr/share/locale/'
#     GNOME_TERMINAL_SCREEN = '/org/gnome/Terminal/screen/29e7c30f_6c9a_4522_915c_0e59e733ffc8'
#     DEFAULTS_PATH = '/usr/share/gconf/ubuntu.default.path'
#     PWD = '/home/andre/workspace/EtherollApp'
#     HOME = '/home/andre'
#     TEXTDOMAIN = 'im-config'
#     SSH_AGENT_PID = '1963'
#     QT_ACCESSIBILITY = '1'
#     LIBVIRT_DEFAULT_URI = 'qemu:///system'
#     TMUX = '/tmp/tmux-1000/default,2469,0'
#     XDG_SESSION_TYPE = 'x11'
#     XDG_DATA_DIRS = '/usr/share/ubuntu:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop'
#     XDG_SESSION_DESKTOP = 'ubuntu'
#     GJS_DEBUG_OUTPUT = 'stderr'
#     GTK_MODULES = 'gail:atk-bridge'
#     WINDOWPATH = '2'
#     VTE_VERSION = '5202'
#     TERM = 'screen'
#     SHELL = '/bin/bash'
#     QT_IM_MODULE = 'ibus'
#     XMODIFIERS = '@im=ibus'
#     IM_CONFIG_PHASE = '2'
#     NVM_BIN = '/home/andre/bin/nvm-0.33.11/versions/node/v10.13.0/bin'
#     XDG_CURRENT_DESKTOP = 'ubuntu:GNOME'
#     GPG_AGENT_INFO = '/run/user/1000/gnupg/S.gpg-agent:0:1'
#     GNOME_TERMINAL_SERVICE = ':1.76'
#     TMUX_PANE = '%132'
#     XDG_SEAT = 'seat0'
#     SHLVL = '2'
#     LANGUAGE = 'en_US.UTF-8'
#     GDMSESSION = 'ubuntu'
#     GNOME_DESKTOP_SESSION_ID = 'this-is-deprecated'
#     LOGNAME = 'andre'
#     DBUS_SESSION_BUS_ADDRESS = 'unix:path=/run/user/1000/bus'
#     XDG_RUNTIME_DIR = '/run/user/1000'
#     XAUTHORITY = '/run/user/1000/gdm/Xauthority'
#     XDG_CONFIG_DIRS = '/etc/xdg/xdg-ubuntu:/etc/xdg'
#     PATH = '/home/andre/.buildozer/android/platform/apache-ant-1.9.4/bin:/home/andre/bin/node_modules/.bin:/home/andre/.local/bin:/home/andre/bin:/home/andre/bin/nvm-0.33.11/versions/node/v10.13.0/bin:/home/andre/bin/node_modules/.bin:/home/andre/.local/bin:/home/andre/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin'
#     GJS_DEBUG_TOPICS = 'JS ERROR;JS LOG'
#     SESSION_MANAGER = 'local/localhost.localdomain:@/tmp/.ICE-unix/1672,unix/localhost.localdomain:/tmp/.ICE-unix/1672'
#     LESSOPEN = '| /usr/bin/lesspipe %s'
#     GTK_IM_MODULE = 'ibus'
#     LC_TIME = 'en_GB.UTF-8'
#     OLDPWD = '/home/andre/workspace/Elements/evaluation-white-label'
#     _ = '/home/andre/.local/bin/buildozer'
#     PACKAGES_PATH = '/home/andre/.buildozer/android/packages'
#     ANDROIDSDK = '/home/andre/.buildozer/android/platform/android-sdk'
#     ANDROIDNDK = '/home/andre/.buildozer/android/platform/android-ndk-r19b'
#     ANDROIDAPI = '27'
#     ANDROIDMINAPI = '21'
# 
# Buildozer failed to execute the last command
# The error might be hidden in the log above this error
# Please read the full log, and search for it before
# raising an issue with buildozer itself.
# In case of a bug report, please add a full log with log_level = 2
```
```

**Comment by Julian-O** [comment]
@npetitclerc Did cleaning the Buildozer directory solve this problem?
